### PR TITLE
Updates to the solar device to handle configuration-related telegrams

### PR DESCRIPTION
**Pull request by hpanther**

### CODE REVIEW

File: src/devices/solar.cpp [hunk 1]
```diff
@@ -406,4 +486,15 @@ void Solar::process_ISM1Set(std::shared_ptr<const Telegram> telegram) {
     changed_ |= telegram->read_value(setpoint_maxBottomTemp_, 6);
 }
 
+bool Solar::set_SM100Tank1MaxTemp(const char * value, const int8_t id) {
+    int temp;
```
Comment (proddy) [inline src/devices/solar.cpp:490]: the types are important as they're used in the Template to render the version. Please use int8_t, uin8_t, init16_t etc

File: src/devices/solar.cpp [hunk 2]
```diff
@@ -406,4 +486,15 @@ void Solar::process_ISM1Set(std::shared_ptr<const Telegram> telegram) {
...
+    if(! Helpers::value2number(value, temp)) return false;
+    tank1MaxTempCurrent_=temp;
+    unsigned char transport_temp=(unsigned char) temp;
```
Comment (proddy) [inline src/devices/solar.cpp:493]: transport_temp can be uint8_t ?

File: src/devices/solar.cpp [hunk 3]
```diff
@@ -270,6 +283,63 @@ void Solar::process_SM10Monitor(std::shared_ptr<const Telegram> telegram) {
     changed_ |= telegram->read_value(pumpWorkMin_, 8, 3);
 }
 
+/*
+ * process_SM100SystemConfig - type 0x0358 EMS+ - for MS/SM100 and MS/SM200
```
Comment (proddy) [inline src/devices/solar.cpp:287]: can you provide an example telegram so I can add to the offline test suite?

File: src/devices/solar.cpp [hunk 4]
```diff
@@ -270,6 +283,63 @@ void Solar::process_SM10Monitor(std::shared_ptr<const Telegram> telegram) {
...
+void Solar::process_SM100SystemConfig(std::shared_ptr<const Telegram> telegram) {
+    changed_ |= telegram->read_value(heatTransferSystem_  , 5, 1);
+    changed_ |= telegram->read_value(externalTank_        , 9, 1);
+    changed_ |= telegram->read_value(thermalDisinfect_    , 10, 1);
+    changed_ |= telegram->read_value(heatMetering_        , 14, 1);
+    changed_ |= telegram->read_value(solarIsEnabled_      , 19, 1);
+}
+
+/*
+ * process_SM100SolarCircuitConfig - type 0x035A EMS+ - for MS/SM100 and MS/SM200
```
Comment (proddy) [inline src/devices/solar.cpp:298]: can you provide an example telegram so I can add to the offline test suite?

**Review by proddy — COMMENTED** [review]
nicely implemented, great to see how you've understood the code logic (which sometimes isn't that logically). I made a few comments below. Be careful on the assignment declarators as they're used in the helper code to correctly render the value.

Also don't you want to have these values in the MQTT payload?

File: src/devices/solar.cpp (hunk 1 above)
Comment (hpanther) [reply src/devices/solar.cpp:490]: Interestingly, when I replace int temp; with int8_t temp; I get a compile error. Helpers::value2number takes an actual int.

**Comment by hpanther** [comment]
Continue tracking this in https://github.com/proddy/EMS-ESP/pull/613

**Pull request closed without merging**